test/data/v3/basic/tests/deployment_test.yaml

```diff
@@ -62,6 +62,9 @@ tests:
           of: extensions/v1beta1
       - hasDocuments:
           count: 2
+      - hasDocuments:
+          count: 1
+          filterAware: true
```

Comment:
Maybe a name should be more self aware, 

`filterSelector`, `selectoAware` or similar

Reply:
We could release and w8 for someone to suggest

Options..., but personally not sure if I see a strong candidate

1. filterApplied: Indicates that the count is aware of or applies a certain filter.
2. useFilter: Suggests that a filter should be used with assertion
3. filterEnabled: Conveys that filtering is enabled for this assertion

Reply:
Kept the naming for now, and we will see if there will be better names.